app/scripts/lib/app-start.js

```diff
@@ -482,18 +486,26 @@ function (
 
     /**
      * Sets a UUID value that is unrelated to any account information.
-     * This value is useful to determine if the logged out user qualifies for ab testing or metrics.
+     * This value is useful to determine if the logged out user qualifies
+     * for A/B testing or metrics.
      */
     initializeUuid: function () {
       var storage = Storage.factory('localStorage', this._window);
 
-      // we reuse the same uuid if it is available in local storage.
-      if (storage.get('uuid')) {
+      // Reuse the same uuid if it is available in local storage.
+      if (this._searchParam('resume')) {
+        var resumeParam = this._searchParam('resume');
+        var resumeToken = new ResumeToken(ResumeToken.parse(resumeParam));
+        this._uuid = resumeToken.get('uuid');
+      } else if (storage.get('uuid')) {
```

Comment:
We should not use an `else if` here but an `if (! this._uid && storage.get('uuid'))` to catch the transition case where a user may have a uuid in localStorage from old code, but not in the ResumeToken since the `uuid` is only passed starting with this PR.